### PR TITLE
Fix part of #103: Add lint tests

### LINKED ISSUE #103: Add lint tests
Enforce lint tests in oppiabot repo.

### PR DESCRIPTION
**Pull request by anandwana001**
## Explanation

Fix part of #103 
SetUp ESLint

### For Reviewers
I had setup eslint and fix .js files inside `__test__` and `actions` package. 

Fixed below files:
 - `actions/main.js`
 - `actions/build.js`
 - `actions/src/dispatcher.js`
 - `actions/src/issues/checkIssueLabels.js`

## Checklist
- [ ] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [ ] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

Adding @jameesjohn for initial Review

### CODE REVIEW

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
Why do we have two eslint configs (.eslindrc and eslintrc.json), please merge these together. Also, lets keep the indent of 2 spaces (we use 2 spaces in https://github.com/oppia/oppia). And I just realized that we do not have a CI job for checking eslint, I think it should be included as part of this PR so that PRs not passing eslint would fail.

File: .eslintrc [hunk 1]
```diff
@@ -5,6 +5,11 @@
   "parserOptions": {
     "ecmaVersion": 8,
   },
+  "overrides": [{
+    // We exclude test files from dependency injection format checks because
```
Comment (jameesjohn) [inline .eslintrc:9]: What do you mean by dependency injection format checks?

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        "quotes": [
+            "error",
+            "single"
+        ],
+        "semi": [
+            "error",
+            "always"
+        ]
+    }
+}
```
Comment (jameesjohn) [inline .eslintrc.json:29]: End with a new line.
Also, as @vojtechjelinek said, we should use either `.eslintrc` or this, but not both.

File: __tests__/index.test.js [hunk 3]
```diff
@@ -16,8 +16,8 @@
 // const index = require('../index')
 
 test('that we can run tests', () => {
-  // your real tests go here
-  expect(1 + 2 + 3).toBe(6);
+    // your real tests go here
+    expect(1 + 2 + 3).toBe(6);
 });
 
 // For more information about testing with Jest see:
```
Comment (jameesjohn) [inline __tests__/index.test.js:23]: I think we can delete this file as it is not being used.
Tests are written in the `/spec` directory.

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks, @anandwana001. I left some comments. Do well to request a re-review after addressing them.

**Comment by vojtechjelinek** [comment]
@anandwana001 Do you need any help here?

**Comment by anandwana001** [comment]
> @anandwana001 Do you need any help here?
> 

I will be pushing suggested changes, delayed due to weekend.

File: .eslintrc (hunk 1 above)
Comment (anandwana001) [reply .eslintrc:9]: Following from the Oppia web
https://github.com/oppia/oppia/blob/develop/.eslintrc#L17

File: .eslintrc.json (hunk 2 above)
Comment (anandwana001) [reply .eslintrc.json]: Done

File: __tests__/index.test.js (hunk 3 above)
Comment (anandwana001) [reply __tests__/index.test.js]: Done

**Comment by anandwana001** [comment]
@jameesjohn 

1. Should we include [these](https://github.com/oppia/oppia/tree/develop/scripts/linters/custom_eslint_checks/rules) eslint rules from oppia-web?

2. Should we follow the [same](https://github.com/oppia/oppia/blob/develop/scripts/run_custom_eslint_tests.py) from oppia-web for adding ci check?

File: .eslintrc (hunk 1 above)
Comment (jameesjohn) [reply .eslintrc:9]: Hmmm...
If so, I don't think we need the `overrides` section.

File: .github/workflows/eslint_tests.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: CI ESLint
```
Comment (jameesjohn) [inline .github/workflows/eslint_tests.yml:1]: CI ESLint --> CI Lint

File: actions/src/dispatcher.js [hunk 5]
```diff
@@ -27,9 +27,9 @@ module.exports = {
     const checkEvent = `${event}_${action}`;
     const repoName = context.payload.repository.name.toLowerCase();
     const checksWhitelist = constants.getChecksWhitelist();
-    if (checksWhitelist.hasOwnProperty(repoName)) {
+    if (Object.prototype.hasOwnProperty.call(checksWhitelist, repoName)) {
```
Comment (jameesjohn) [inline actions/src/dispatcher.js:30]: Why is this changed?

**Review by jameesjohn — CHANGES_REQUESTED** [review]
Thanks, @anandwana001. I left a few more comments.

**Comment by jameesjohn** [comment]
> Should we include these eslint rules from oppia-web?

I don't think we need those.
> Should we follow the same from oppia-web for adding ci check?

Nope. The GitHub action should be just fine.

File: actions/src/issues/checkIssueLabels.js [hunk 6]
```diff
@@ -30,7 +30,7 @@ const checkLabels = async () => {
   const user = context.payload.sender.login;
 
   if (
-      label.name === GOOD_FIRST_LABEL &&
+    label.name === GOOD_FIRST_LABEL &&
       !whitelist.goodFirstIssue.includes(user)) {
```
Comment (vojtechjelinek) [inline actions/src/issues/checkIssueLabels.js:34]: I would do something like
```suggestion
    label.name === GOOD_FIRST_LABEL &&
    !whitelist.goodFirstIssue.includes(user)
  ) {
```

**Review by vojtechjelinek — APPROVED** [review]
LGTM! Just one small nit.

File: actions/src/dispatcher.js (hunk 5 above)
Comment (anandwana001) [reply actions/src/dispatcher.js:30]: As per the eslint, if we call `hasOwnProperty` from any object directly, this might lead to unsafe way. So, using `Object.prototype` is a safe way to prevent such errors or bug might occur in future.  

It gives the following lint error:

```
30:25  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins
32:18  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins
```

File: .eslintrc (hunk 1 above)
Comment (anandwana001) [reply .eslintrc]: Done

File: .github/workflows/eslint_tests.yml (hunk 4 above)
Comment (anandwana001) [reply .github/workflows/eslint_tests.yml]: Done

File: actions/src/issues/checkIssueLabels.js (hunk 6 above)
Comment (anandwana001) [reply actions/src/issues/checkIssueLabels.js]: Done, 
I forget to do the file reformat thing. Thanks for reminding.

File: actions/src/dispatcher.js (hunk 5 above)
Comment (jameesjohn) [reply actions/src/dispatcher.js:30]: Got it. 
Thanks for the explanation.

**Review by jameesjohn — APPROVED** [review]
LGTM!
Thanks, @anandwana001!
